server.js

```diff
@@ -1,30 +1,192 @@
+  - **Body:** A JSON object containing an array of titles from the specified country.
+  
+  ## Error Responses
+  
+  - **Status Code:** 404 Not Found
+  - **Body:** An error message indicating that the requested resource was not found.
+  
+  - **Status Code:** 500 Internal Server Error
+  - **Body:** An error message indicating that there was an internal server error.
+  `;
+  const htmlDocumentation = md.render(documentation);
+  res.send(htmlDocumentation);
+});
+
+// Endpoint for all the data from netflixData
+app.get('/titles', (req, res) => {
+  res.json(netflixData);
+  console.log(`Showing netliixdata`);
+});
+
+// Endpoint to fetch a specific title by its show_id
+app.get('/titles/:id', (req, res) => {
+  const id = req.params.id;
+  const titleId = netflixData.find((titleId) => titleId.show_id === +id);
+
+    // Respond with the details of the specific title otherwise send 404 status
+  if (titleId) {
+    res.json(titleId);
+  } else {
+    res.status(404).send('No title was found, please try again!');
+  }
+});
+
+// Endpoint to fetch titles in a specific category
+app.get('/titles/categories/:category', (req, res) => {
+  const requestedCategory = req.params.category.toLowerCase();
+  const mediaInCategory = netflixData.filter(media => media.listed_in.toLowerCase().includes(requestedCategory));
+  const mediaTitles = mediaInCategory.map(media => media.title);
+
+    // Respond with the titles in the specified category, otherwise send 404 status
+    if (mediaTitles.length === 0) {
+      res.status(404).send('Category not found, please try again!');
+    } else {
+      res.json({ mediaInCategory: mediaTitles });
+    }
+});
+
+// Endpoint to fetch titles of a specific type (movie or TV show)
+app.get('/titles/types/:type', (req, res) => {
+  const requestedType = req.params.type.toLowerCase();
+  const movieOrTvshow = netflixData.filter(typeOfMedia => typeOfMedia.type.toLowerCase().includes(requestedType));
+  const mediaTitlesType = movieOrTvshow.map(typeOfMedia => typeOfMedia.title);
+
+    // Respond with the titles of the specified type, otherwise send 404 status
+  if (mediaTitlesType.length === 0){
+    res.status(404).send('That is not an available media type, try movie or tv show please.');
+  } else {
+  res.json({ movieOrTvshow: mediaTitlesType });
+  }
+});
+
+// Endpoint to fetch titles from a specific country
+app.get('/titles/countries/:country', (req, res) => {
+  const requestedCountry = req.params.country.toLowerCase();
+  const mediaFromCountry = netflixData.filter(mediaCountry => mediaCountry.country.toLowerCase().includes(requestedCountry));
+  const mediaTitlesCountry = mediaFromCountry.map(mediaCountry => mediaCountry.title);
+
+    // Respond with the titles from the specified country, otherwise send 404 status
+   if (mediaTitlesCountry.length === 0){
+    res.status(404).send('That is not an available country, please try again!')
```

Comment:
I guess the entirely proper error message here would be "There is no available media from the requested country", as the if-statement checks if the requested data from country is zero, and not if the country is available or not.